build_tools/cmake/iree_filter_test.cmake

```
@@ -0,0 +1,65 @@
+# Copyright 2023 The IREE Authors
+#
+# Licensed under the Apache License v2.0 with LLVM Exceptions.
+# See https://llvm.org/LICENSE.txt for license information.
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+include(CMakeParseArguments)
+
+function(iree_filter_test)
+  cmake_parse_arguments(
+    _FUNC
+    ""
+    "RESULT_VAR_ENABLED;SIZE;DRIVER"
+    "LABELS"
+    ${ARGN}
+  )
+  # Default to excluding; then exclusion cases are implemented as early-returns;
+  # then at the end we will override that to TRUE.
+  set("${_FUNC_RESULT_VAR_ENABLED}" FALSE PARENT_SCOPE)
+  # Exclude hostonly tests when cross-compiling.
+  if(CMAKE_CROSSCOMPILING AND "hostonly" IN_LIST _FUNC_LABELS)
+    return()
+  endif()
+  # Sanitizer-based exclusions.
+  # CUDA is considered incompatible with all sanitizers.
+  if(IREE_ENABLE_ASAN)
```

nit: can you add newlines between each section? That could make the file easier to scan (file size isn't a concern, since this is the only function in the file)

```suggestion
  # Exclude hostonly tests when cross-compiling.
  if(CMAKE_CROSSCOMPILING AND "hostonly" IN_LIST _FUNC_LABELS)
    return()
  endif()
  
  # Sanitizer-based exclusions.
  # CUDA is considered incompatible with all sanitizers.
  if(IREE_ENABLE_ASAN)
```